include/oc_network_monitor.h

```diff
@@ -21,45 +21,84 @@
 #ifndef OC_NETWORK_HELPERS_H
 #define OC_NETWORK_HELPERS_H
 
+#include "oc_export.h"
 #include "oc_network_events.h"
 #include "oc_session_events.h"
+#include "util/oc_compiler.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
-  @brief Add the callback to receive change notifications for
-   Network interface event.
-  @param cb  The callback to be added. Must not be NULL.
-  @return 0 if success. -1 return if error occur.
-*/
+ * @brief Add the callback to receive change notifications for Network interface
+ * event.
+ * @param cb  The callback to be added. Must not be NULL.
+ * @return 0 on success
+ * @return -1 on error
+ */
 int oc_add_network_interface_event_callback(interface_event_handler_t cb);
 
 /**
-  @brief Remove the callback to receive change notifications for
-   Network interface event.
-  @param cb  The callback to be removed. Must not be NULL.
-  @return 0 if success. -1 return if error occur.
-*/
+ * @brief Remove the callback to receive change notifications for Network
+ * interface event.
+ * @param cb  The callback to be removed. Must not be NULL.
+ * @return 0 on success
+ * @return -1 on error
+ */
 int oc_remove_network_interface_event_callback(interface_event_handler_t cb);
 
 /**
-  @brief Add the callback to receive change notifications for
-   Network interface event.
-  @param cb  The callback to be added. Must not be NULL.
-  @return 0 if success. -1 return if error occur.
-*/
-int oc_add_session_event_callback(session_event_handler_t cb);
+ * @brief Add the callback to receive session event notifications.
+ *
+ * @param cb  The callback to be added. Must not be NULL.
+ * @return 0 on success
+ * @return -1 on error
+ *
+ * @deprecated replaced by oc_add_session_event_callback_v1 in v2.2.5.4
+ */
+OC_API
+int oc_add_session_event_callback(session_event_handler_t cb) OC_DEPRECATED(
+  "replaced by oc_add_session_event_callback_v1, deprecated in v2.2.5.4");
 
 /**
-  @brief Remove the callback to receive change notifications for
-   Network interface event.
-  @param cb  The callback to be removed. Must not be NULL.
-  @return 0 if success. -1 return if error occur.
-*/
+ * @brief Add the callback to receive session event notifications.
+ *
+ * @param cb The callback to be added (cannot be NULL).
+ * @param user_data user data passed to the callback when invoked
+ * @return 0 on success
+ * @return -1 on error
+ */
+OC_API
+int oc_add_session_event_callback_v1(session_event_handler_v1_t cb,
```

Comment:
return bool instead of int ?

Reply:
wanted to keep it consistent with `oc_add_session_event_callback` and `oc_remove_session_event_callback` which return error value.